core/templates/dev/head/services/PromoBarService.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service Promo bar.
+ */
+
+oppia.factory('PromoBarService', [
+  '$http', 'ENABLE_PROMO_BAR',
+  function($http, ENABLE_PROMO_BAR) {
+    return {
+      getPromoBarData: function() {
+        var promoBarData = {
+          promoBarEnabled: false,
+          promoBarMessage: ''
+        };
+
+        return $http.get('/promo_bar_handler', {}).then(
+          function(response) {
+            if (ENABLE_PROMO_BAR) {
```

Comment:
One more thing, we shouldn't call the `$http.get` when `ENABLE_PROMO_BAR` is false. Move the if outside of `function(response) {…}` and call `return $q.resolve(promoBarData);` when `ENABLE_PROMO_BAR` is false

Reply:
Done. I didn't know about `$q.resolve` earlier. Thanks for introducing me to this new concept.